test/e2e/storage/persistent_volumes-local.go

```diff
@@ -53,98 +60,309 @@ const (
+		})
+	})
+
+	Context("when two pods request one prebound PVC at the same time", func() {
+		var (
+			testVol *localTestVolume
+		)
+
+		BeforeEach(func() {
+			testVol = setupLocalVolumePVCPV(config, node0)
+		})
+
+		AfterEach(func() {
+			cleanupLocalVolume(config, testVol)
+		})
+
+		It("should be able to mount volume in two pods at the same time, write from pod1, and read from pod2", func() {
+			By("Creating pod1 to write to the PV")
+			pod1, pod1Err := createLocalPod(config, testVol)
+			Expect(pod1Err).NotTo(HaveOccurred())
+
+			framework.ExpectNoError(framework.WaitForPodRunningInNamespace(config.client, pod1))
+			pod1NodeName, pod1NodeNameErr := podNodeName(config, pod1)
+			Expect(pod1NodeNameErr).NotTo(HaveOccurred())
+			framework.Logf("Pod1 %q created on Node %q", pod1.Name, pod1NodeName)
+			Expect(pod1NodeName).To(Equal(node0.Name))
+
+			By("Creating pod2 to read from the PV")
+			pod2, pod2Err := createLocalPod(config, testVol)
+			Expect(pod2Err).NotTo(HaveOccurred())
+
+			framework.ExpectNoError(framework.WaitForPodRunningInNamespace(config.client, pod2))
+			pod2NodeName, pod2NodeNameErr := podNodeName(config, pod2)
+			Expect(pod2NodeNameErr).NotTo(HaveOccurred())
+			framework.Logf("Pod2 %q created on Node %q", pod2.Name, pod2NodeName)
+			Expect(pod2NodeName).To(Equal(node0.Name))
+
+			writeCmd, readCmd := createWriteAndReadCmds(volumeDir, testFile, testVol.hostDir /*writeTestFileContent*/)
+
+			By("Writing in pod1")
+			podRWCmdExec(pod1, writeCmd)
+			By("Reading in pod2")
+			readOut := podRWCmdExec(pod2, readCmd)
+
+			Expect(readOut).To(ContainSubstring(testVol.hostDir)) /*aka writeTestFileContents*/
+
+			By("Deleting pod1")
+			framework.DeletePodOrFail(config.client, config.ns, pod1.Name)
+			By("Deleting pod2")
+			framework.DeletePodOrFail(config.client, config.ns, pod2.Name)
+		})
+	})
+
+	Context("when using local volume provisioner", func() {
+		var (
+			volumePath string
+		)
+
+		BeforeEach(func() {
+			setupLocalVolumeProvisioner(config, node0)
+			volumePath = path.Join(
+				hostBase, discoveryDir, fmt.Sprintf("vol-%v", string(uuid.NewUUID())))
+		})
+
+		AfterEach(func() {
+			cleanupLocalVolumeProvisioner(config, node0, volumePath)
+		})
+
+		It("should create and recreate local persistent volume", func() {
+			By("Creating bootstrapper pod to start provisioner daemonset")
+			createBootstrapperPod(config)
+			kind := schema.GroupKind{"extensions", "DaemonSet"}
+			framework.WaitForControlledPodsRunning(config.client, config.ns, daemonSetName, kind)
+
+			By("Creating a directory under discovery path")
+			mkdirCmd := fmt.Sprintf("mkdir %v", volumePath)
+			runLocalUtil(config, node0.Name, mkdirCmd, hostBase)
+			err := waitForLocalPersistentVolume(config.client, volumePath)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Create a persistent volume claim for local volume: the above volume will be bound.
+			// Note: make sure no other local PVs are available; otherwise, the test can be flaky.
```

Comment:
oh, this is a left-over from e2e test in provisioner, we already use local volume specific storageclass here. I'll remove it.